src/domain.rs

```diff
@@ -350,10 +355,18 @@ impl<F> Domain<F> {
     fn check_threshold_and_reclaim(&self) -> usize {
         let mut rcount = self.check_count_threshold();
         if rcount == 0 {
-            rcount = self.check_due_time();
```

Comment:
Do you think it would be nicer to write the body of this if as:
```rust
...
rcount = self.check_due_time();
if rcount == 0 {
   // TODO: Implement some kind of mock time for no_std.
   // Currently we reclaim only based on rcount on no_std
   #[cfg(feature = "std")]
   rcount = self.check_due_time();

   if rcount == 0 {
       return 0;
   }
}
```
The compiler will almost certainly optimize out the inner `if` for `no_std`, and it makes the conditional part only one line. Still looks a little goofy so I can't decide.

Reply:
Yeah, I like that!